logging/logfile.go

```diff
@@ -45,22 +48,112 @@ type LogFile struct {
 	acquire sync.Mutex
 }
 
-func (l *LogFile) fileNamePattern() string {
+// Splits a full filename ("consul.log") into the name without extension ("consul")
+// and the extension (".log"). If the full filename has no extension, the ".log"
+// extension will be returned.
+func splitExtensionFromFileName(fileName string) (fileNameWithoutExt string, fileExt string) {
 	// Extract the file extension
-	fileExt := filepath.Ext(l.fileName)
+	fileExt = filepath.Ext(fileName)
+
 	// If we have no file extension we append .log
 	if fileExt == "" {
 		fileExt = ".log"
 	}
+
 	// Remove the file extension from the filename
-	return strings.TrimSuffix(l.fileName, fileExt) + "-%s" + fileExt
+	fileNameWithoutExt = strings.TrimSuffix(fileName, fileExt)
+
+	return
 }
 
-func (l *LogFile) openNew() error {
-	fileNamePattern := l.fileNamePattern()
+// Based on the configured filename, generates an output filename with the
+// following characteristics:
+// - the ".log" extension is appended if the configured filename has no extension
+// - a Unix timestamp is appended before the extension unless the configured
+//   filename is a template that uses ${timestamp-human} or ${timestamp-unix}
+// - occurrences of ${timestamp-human} are replaced with the
+//   timestampStringHumanReadable input
+// - occurrences of ${timestamp-unix} are replaced with the
+//   timestampStringUnix input
+func (l *LogFile) getGenericFileName(timestampStringUnix string, timestampStringHumanReadable string) (string, error) {
+	fileNameWithoutExt, fileExt := splitExtensionFromFileName(l.fileName)
+
+	parsedFileNameWithoutExtNode, err := hil.Parse(fileNameWithoutExt)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse log filename template: %s", err)
+	}
+
+	// Build HIL evaluation context that maps template fields to values
+	nodeEvalConfig := &hil.EvalConfig{
+		GlobalScope: &ast.BasicScope{
+			VarMap: map[string]ast.Variable{
+				"timestamp-human": {
+					Type:  ast.TypeString,
+					Value: timestampStringHumanReadable,
+				},
+				"timestamp-unix": {
+					Type:  ast.TypeString,
+					Value: timestampStringUnix,
```

Comment:
I agree with the idea of using standard Go semantics, but I think in this case it sets the wrong expectations.

For example: 
```
time.Stamp      = "Jan _2 15:04:05"
time.StampNano  = "Jan _2 15:04:05.000000000"
```

Both look very different than the format currently in the code, which would be like `1629417801659820807` (just an integer).

I have a similar concern about `date.rfc3339` and `date.iso8601`, neither of which is similar to the current format generated by the code (e.g., `2021-08-19_20-13-08-322108785`).

I think the result needs to match any existing semantics we rely on. And I don't think we want to match the existing semantics in a filename (special characters, harder to sort files by filename).

What do you think?

Reply:
Ah ya, sorry my suggestion was misleading. I was thinking that `time.UnixNano()` should return the right value for `date.timestamp`, and maybe a custom format for  the human readable one. Since it's a custom format I guess using the ISO and RFC names is not ideal, so maybe ``date.human` would be good.

Reply:
Preference on
- `date.human` + `date.unix` vs
- `time.human` and `time.unix`?

I've always found it strange that `date` is used to give date and time (e.g., `date` unix command, `time.Date()` Golang object), which is why I was initially leaning towards `time` or `timestamp`. But I don't feel strongly either way.

(Or let me know if I misunderstood your comment and you had something else in mind)